WIP Reusable `ProjectComponent.vue` from `Project.vue`

wackerow opened the pull request:
- Interfaced form data with Project data type
- Extracts ProjectComponent from the Project view
- Component takes a prop of data type "Project", which is used when interacting with the smart contracts. 
- Allows simpler reuse for both displaying the project details live, and also for applicants to preview their listing in a more realistic manner before submitting.

_rough draft_

wackerow committed:
project display refactor draft

WIP. Interfaced form data with Project data type. Extracts Project component that could be reused with contract project data, and also form summary page.

wackerow committed:
Merge branch 'develop' into project-component

wackerow committed:
typo fix

wackerow committed:
adds markdown for description

wackerow committed:
adds markdown to description of ProjectComponent

renames `data` to `project`

wackerow committed:
ts linting

samajammin commented on vue-app/src/components/ProjectComponent.vue:
```diff
@@ -0,0 +1,274 @@
+import { Project } from '@/api/projects'
+import { markdown } from '@/utils/markdown'
+
+@Component export default class ProjectComponent extends Vue {
```
Just an opinion but `ProjectProfile` instead of `ProjectComponent`? No need to call it a component 😄

wackerow replied:
Yeah that works, I just didn't want to use `Project` again since we already have `/views/Project.vue`, I'll change it

wackerow committed:
updates registry for Urls instead of hashes

wackerow committed:
project grid layout

pulls banner image out of ProjectComponent (could now be renamed to ProjectDetails)

wackerow committed:
IpfsForm updates

wackerow committed:
summary page conditional layouts

allows toggling preview display

wackerow committed:
dummy form data for dev

wackerow committed:
ts linting

wackerow committed:
ProjectProfile rename

wackerow committed:
project-profile rename

samajammin commented on vue-app/src/components/IpfsForm.vue:
```diff
@@ -119,6 +113,10 @@ export default class IpfsForm extends Vue {
     }
   }
+
+  get bannerUrl(): string {
```
`imageUrl`?

samajammin commented:
@wackerow this is looking good but curious to get @ryancreatescopy's opinion. Having this render right in the form layout doesn't make it obvious to me that we're previewing what the actually project page is going to look like. I wonder how was could make that more clear to the user (perhaps some copy is all it needs).

Also should we somehow display the links widget?
![Image 2021-04-29 at 5 38 43 PM](https://user-images.githubusercontent.com/8097623/116634837-0b5bc080-a912-11eb-85a4-e0a2830ebcf9.jpg)

samajammin committed:
Add markdown component & markdown header styles

samajammin committed:
Rename IpfsImageUpload component

samajammin committed:
Update markdown styles

samajammin committed:
Add markdown previews for textarea inputs

samajammin committed:
Comment out dummy data
